app/models/genre.rb

```diff
@@ -3,4 +3,9 @@ class Genre < ApplicationRecord
   has_many :songs, through: :song_genres
 
   validates_presence_of :name
+
+  def avg_rating
+  
+    Genre.joins(:songs).average(:rating)
```

Comment:
since you have an instance of a genre already, and this is supposed to give an average of just THIS genre's songs, the code you've written above will actually provide an average of all songs no matter their genre.
Correct code would have just been `songs.average(:rating)` here, since we can utilize the has_many relationship established in your model, which gives us that ACtiveRecord association, and then chain on additional AR commands.

Reply:
Gotcha.  I see that now.  Use the specific Genre's songs to query the average ratings instead of the entire class linked to all songs.